includes/classes/db/mysql/query_factory.php

```diff
@@ -645,17 +658,37 @@ protected function show_error()
         if (!headers_sent()) {
             header("HTTP/1.1 503 Service Unavailable");
         }
-        if ($this->error_number == 0 && $this->error_text == DB_ERROR_NOT_CONNECTED && file_exists(FILENAME_DATABASE_TEMPORARILY_DOWN)) {
-            include(FILENAME_DATABASE_TEMPORARILY_DOWN);
+
+        if (!defined('FILENAME_DATABASE_TEMPORARILY_DOWN')) {
+            define('FILENAME_DATABASE_TEMPORARILY_DOWN', DIR_FS_CATALOG . '/nddbc.html');
+        }
+        if (file_exists(FILENAME_DATABASE_TEMPORARILY_DOWN)) {
+            if (($this->error_number == 0 && $this->error_text == DB_ERROR_NOT_CONNECTED)
+                || $this->error_number == 2002)
+            {
+                include(FILENAME_DATABASE_TEMPORARILY_DOWN);
+            }
+        }
+
+        // suppress backtrace for MariaDB connection errors: not logging these because they usually come hundreds at a time
+        if ($this->error_number == 2002 || defined('DIR_FS_INSTALL')) {
+            return;
         }
+
+        // display error details if appropriate
         echo '<div class="systemError">';
         if (defined('STRICT_ERROR_REPORTING') && STRICT_ERROR_REPORTING == true) {
             echo $this->error_number . ' ' . $this->error_text;
             echo '<br>in:<br>[' . (strstr($this->zf_sql, 'db_cache') ? 'db_cache table' : $this->zf_sql) . ']<br>';
         } else {
             echo 'WARNING: An Error occurred, please let us know!';
         }
+        if (defined('IS_ADMIN_FLAG') && IS_ADMIN_FLAG == true) {
```

Comment:
Use `===`

Reply:
That can be done separately. I didn't add/change this code in this PR, merely relocated it.